vfs/vfscache/item.go

```diff
@@ -767,6 +806,165 @@ func (item *Item) remove(reason string) (wasWriting bool) {
 	return item._remove(reason)
 }
 
+// GetCache returns the parent cache
+func (item *Item) GetCache() (cache *Cache) {
+	item.mu.Lock()
+	defer item.mu.Unlock()
+	return item.c
+}
+
+// Reset waits for cache read activities to subside, removes the cache file, then reopen it.
+// Used when cache space runs out.  Since Reset is called by the cache cleaner thread, there
+// is no direct way to return the error to the io threads.  Set item.fd to nil upon internal
+// errors so that the io threads will return internal errors seeing a nill fd.
+func (item *Item) Reset() {
+	var (
+		err error
+	)
+	item.mu.Lock()
+	defer item.mu.Unlock()
+
+	if item.opens == 0 {
+		return
+	}
+
+	item.beingReset = true
+	if item.pendingReads > 0 {
+		for {
+			if item.pendingReads == 0 {
+				break
+			}
+			item.cond.Wait()
+		}
+	}
+
+	fso := item.o
+	/*
+		item.mu.Unlock()
+		err := item.Close(nil)
+		if err != nil {
+			fs.Errorf(item.c.fcache, "vfs cache: failed to close item %s: %v", item.name, err)
+		}
+		item.remove("out of space")
+		err = item.Open(fso, true)
+		if err != nil {
+			fs.Errorf(item.c.fcache, "vfs cache: failed to reopen item %s: %v", item.name, err)
+		} */
+	// Close the downloaders
+	// Accumulate and log errors
+	checkErr := func(e error) {
+		if e != nil {
+			fs.Errorf(item.o, "vfs cache: item reset failed: %v", e)
+			if err == nil {
+				err = e
+			}
+		}
+	}
+
+	if downloaders := item.downloaders; downloaders != nil {
+		item.downloaders = nil
+		// FIXME need to unlock to kill downloader - should we
```

Comment:
Mmm, possibly.  The locking is super tricky here. It is really easy to cause a deadlock so if we want to do that then it should be a different commit.

Reply:
Are you referring to the comment in Line 866?  This is actually your comment that got copied from item.close since we are reusing a part of the item.close code.  I do not try to address this comment now. :)